### ISSUE TITLE
Queued points not clearing after successful post

### ISSUE THREAD

**Issue by gloriouslyawkwardlife** [aaronpk/Overland-iOS#166]
I have set up a Google Apps Script to receive the output from the Overland app and create a JSON file in my Google Drive. While the execution is successful and returning the expected `{ "result": "ok"}`, the queued locations are not being cleared after the post.

The script:

```js
function doPost(e) {
  DriveApp.createFile(`${new Date().toISOString()}.overland.json`,e.postData.contents,'application/json');
  var reply = ContentService.createTextOutput();
  reply.append(
    `{
  "result": "ok"
}`);
  return reply;
}
```

**Comment by aaronpk** [comment]
Do you have any way to tell how long this script takes to return the success response? I haven't set any timeouts on the http client and I don't know what the default is, but my guess is it's timing out before the app gets the response.

**Comment by gloriouslyawkwardlife** [comment]
In the Apps Script Executions log it's showing 2.246s for 7,444 queued locations, 1.267s when there were about 4,500. Google Apps Script _always_ returns an HTTP 200 even if there is an error in the script (the response would be an HTML page indicating the error), so I have turned _on_ the "Consider HTTP 2XX Successful" setting.

I can confirm that the payloads are getting through (see the attached screenshot); but the app isn't clearing the queue afterward.  

![Screenshot 2024-10-18 at 9 08 13 AM](https://github.com/user-attachments/assets/0496f601-14af-400d-b923-63cead55effe)

If it helps I am on iOS 18.0.1 on an iPhone 16 Pro Max (1 TB).

I have the same issue when sending it to a CouchDB endpoint as well (CouchDB accepts a simple JSON POST and returns a 200 with the ID of the created record).

**Comment by gloriouslyawkwardlife** [comment]
This just dawned on me—I believe Apps Script does an HTTP redirect; don't know if that affects anything.

**Comment by gloriouslyawkwardlife** [comment]
**UPDATE:** I changed the URL to my CouchDB endpoint (which takes a raw JSON payload) and it worked; I saw the queue count drop. It appears to be something with Google Apps Script.

For what it's worth, I believe GAS does an HTTP redirect to present the response; could the app possibly not be following the redirect?

**Comment by aaronpk** [comment]
Yes, an HTTP redirect means it's first returning a non 200 HTTP code, which the app considers a failure then.
